### DISCUSSION TITLE
SecurityProtocols

### DISCUSSION THREAD

**Discussion by asterismo** [atauenis/webone#91] [Q&A] [answered]
Is this syntaxis ok?

SecurityProtocols=1

I would like the client negotiate with some lower/older sslv3 with the proxy. Is this ok?

**Comment by atauenis** [comment, marked as answer]
Partially correct.

This option configures SSL/TLS versions which WebOne will use to connect to remote servers. The option is set by "OR" operation over .NET internal IDs of TLS versions:

> SecurityProtocols - possible workaround for SSL/TLS errors. It enables or disables particular security protocols. This setting is based on [SecurityProtocolType Enum](https://docs.microsoft.com/en-us/dotnet/api/system.net.securityprotocoltype?view=netframework-4.8) and can be either a single type or an OR-ed combination of them. Default: unset or 4032 (4032 = 192 | 768 | 3072) depending on OS version and MS IE settings.

`SecurityProtocols=1` is not seems to be known by .NET Runtime, but `SecurityProtocols=0` is SystemDefault (and it is equal to 4032 on pre-Windows 10 systems, and to result of `192 | 768 | 3072 | 12288` on all Linux & Mac & Windows 10+), which meaning allow all protocols enabled by OS settings.

**Comment by asterismo** [comment]
Oh, so it does not has nothing to do with client-proxy connection, right? I have an ancient IRC client that has the option of using HTTP proxy, so i tried Webone on it, but it did not worked. The IRC client has SSL option, i think SSLv3 but i'm not sure if it would work with Webone.

**Reply by atauenis** [reply]
No, HTTPS proxies are working different than simple HTTP proxies.

HTTP proxies are just a HTTP server, which can return pages by any URLs (from any remote host). This is how all current versions of WebOne work.

HTTPS proxies are working on TCP level, and can accept any protocol, not only regular HTTP (by default it's SSL, so there's no difference between IRCS and HTTPS - the traffic is always encrypted, and by default the proxy server cannot read or change it without signing with fake certificate). I'm working on implementing such connections support in WebOne, but it's a large and long work.
